File: Core-backend-layer/Programming-fundamentals/Java/DesignPatterns/src/singleton/CarManager.java
package singleton;

// Singleton class to manage Car-related operations
public class CarManager {
    private static CarManager instance;

    // Private constructor to prevent instantiation outside the class
    private CarManager() {
    }

    // Method to get the singleton instance
    public static CarManager getInstance() {
        if (instance == null) {
            instance = new CarManager();
        }
        return instance;
    }

    public void calculateTotalPriceWithTax(Car car, double taxRate) {
        double totalPrice = car.getPrice() * (1 + taxRate / 100);
        System.out.println("Total price for " + car.getName() + " " + car.getModel() + " with " + taxRate + "% tax: $" + totalPrice);
    }

    public void processCarOrder(Car car) {
        System.out.println("Processing order for " + car.getName() + " " + car.getModel() + " with a base price of $" + car.getPrice());
    }
}

File: Core-backend-layer/Programming-fundamentals/Python/DesignPatterns/src/singleton/car_manager.py
# Singleton class to manage car-related operations
class CarManager:
    # Singleton instance variable
    __instance = None

    # Constructor not really needed in Python

    # Static method to get the singleton instance
    @staticmethod
    def get_instance():
        if CarManager.__instance is None:
            CarManager.__instance = CarManager()
        return CarManager.__instance

    def calculate_total_price_with_tax(self, car, tax_rate):
        total_price = car.get_price() * (1 + tax_rate / 100)
        print("Total price for {} {} with {:.2f}% tax: ${:.2f}".format(car.get_name(), car.get_model(), tax_rate, total_price))

    def process_car_order(self, car):
        print("Processing order for {} {} with a base price of ${:.2f}".format(car.get_name(), car.get_model(), car.get_price()))